de/factorization.py

```diff
@@ -60,7 +61,58 @@ def factorizeEstimate(data, tol=1e-9, maxiter=10000):
 
 
 def cellLineFactorization(cellLine):
+    """ Import a cell line, fit the model, and return the result. """
     data, annotation = importLINCS(cellLine)
     w, eta = factorizeEstimate(data)
-
-    return w, eta, annotation
+    return w, eta, annotation[0].tolist()
+
+def cellLineComparison(cellLine1, cellLine2):
+    w1, eta1, annotation1 = cellLineFactorization(cellLine1)
+    w2, eta2, annotation2 = cellLineFactorization(cellLine2)
+
+    line1_as_set = set(annotation1)
+    intersection = line1_as_set.intersection(annotation2)
+    intersection_annotation = list(intersection)
+
+    index_list1 = []
+    index_list2 = []
+
+    for x in intersection_annotation:
+        index_value1 = annotation1.index(x)
+        index_list1.append(index_value1)
+
+    for x in intersection_annotation:
+        index_value2 = annotation2.index(x)
+        index_list2.append(index_value2)
+    
+    index_list1.sort()
+    index_list2.sort()
+    return index_list1, index_list2
+
+def MatrixSubtraction(cellLine1, cellLine2):
+    w1, _, _ = cellLineFactorization(cellLine1)
+    w2, _, _= cellLineFactorization(cellLine2)
+    index_list1, index_list2 = cellLineComparison(cellLine1, cellLine2)
+
+    w1_df = pd.DataFrame(w1)
+    w2_df = pd.DataFrame(w2)
+
+    w1_edited = w1_df.iloc[index_list1, index_list1]
+    w2_edited = w2_df.iloc[index_list2, index_list2]
+
+    w1_final = w1_edited.values
+    w2_final = w2_edited.values
+
+    difference_matrix = w2_final - w1_final
+    norm = np.linalg.norm(difference_matrix)
+    return difference_matrix, norm, w1_final, w2_final
+
+def PearsonWMatrix(cellLine1, cellLine2):
+    _, _, w1, w2 = MatrixSubtraction(cellLine1, cellLine2)
+    pearson = np.corrcoef(w1.flatten(), w2.flatten())
+    return pearson
+
+def SpearmanWMatrix(cellLine1, cellLine2):
+    _, _, w1, w2 = MatrixSubtraction(cellLine1, cellLine2)
```

Comment:
I'm not sure this makes for a good function. It does two things, and calling it isn't much simpler than just running the code it contains.